### PR TITLE
Add GML formatter

### PR DESCRIPTION
**Pull request by linusg**
This adds a GML formatter and uses it in Playground as well as a new clang-format-like Userland program (`gml-format`).

gml-formatter

image

---

I already added `gml-format` to Lagom, so we could easily integrate it into the CI linting step. (in fact this PR was inspired by the GML format bikeshedding happening in another PR)

### CODE REVIEW

**Comment by linusg** [comment]
dammit refined-github stop replacing my PR titles with the first commit message :roll_eyes:

**Comment by alimpfard** [comment]
live formatting wen?

nit: don't mess the input up when it can't be formatted, just don't format at all (or do your best to make things right when there's no ambiguity?)

**Comment by linusg** [comment]
> nit: don't mess the input up when it can't be formatted, just don't format at all (or do your best to make things right when there's no ambiguity?)

the only time it can't be formatted is when the GML is invalid and that's already being checked...

**Comment by alimpfard** [comment]
I assumed that second picture was showing the output (I thought that `<-- missing ...` was being inserted by the formatter), I guess that's part of the input ^^'

**Comment by linusg** [comment]
> I assumed that second picture was showing the output (I thought that <-- missing ... was being inserted by the formatter), I guess that's part of the input ^^'

ohh, yeah it totally looks like an error message, very confusing!
but you're right, it's already part of the input :)

File: DevTools/Playground/main.cpp [hunk 1]
```diff
@@ -300,6 +301,35 @@ int main(int argc, char** argv)
...
+                    GUI::MessageBox::Type::Warning,
+                    GUI::MessageBox::InputType::OKCancel);
+                if (result == GUI::MessageBox::ExecCancel)
+                    return;
+                break;
+            }
+        }
+        auto formatted_gml = GUI::format_gml(source);
+        if (!formatted_gml.is_null()) {
+            editor.set_text(formatted_gml);
```
Comment (alimpfard) [inline DevTools/Playground/main.cpp:323]: Future change idea: Maybe the editor could have a formatter (just like a highlighter and completion provider), and we could later enable "format on save" or "format on change"?

Obviously not a requirements for this PR :P

**Pull request merged**